### PR TITLE
feat: Migrate to functional integration APIs

### PR DESCRIPTION
**Pull request by timfish**
Migrates all the integrations to use the new functional style.

### CODE REVIEW

File: src/main/integrations/electron-breadcrumbs.ts [hunk 1]
```diff
@@ -202,4 +142,57 @@ export class ElectronBreadcrumbs implements Integration {
...
+
+      whenAppReady.then(
+        () => {
+          // We can't access these until app 'ready'
+          if (options.screen) {
+            patchEventEmitter(screen, 'screen', options.screen);
+          }
+
+          if (options.powerMonitor) {
+            patchEventEmitter(powerMonitor, 'powerMonitor', options.powerMonitor);
```
Comment (mydea) [inline src/main/integrations/electron-breadcrumbs.ts:161]: just to be sure, I don't really know how exactly that works for Electron, but is it a problem if we patch this multiple times? Can there even be multiple clients in electron?

Comment (mydea) [reply src/main/integrations/electron-breadcrumbs.ts:161]: 👍

**Review by mydea — APPROVED** [review]
Just skimmed this TBH but overall looks good to me!

**Comment by timfish** [comment]
We've got quite good e2e test coverage. I broke a couple in the process of this PR!

**Pull request merged**